app/views/contexts/context.py

```diff
@@ -19,7 +17,7 @@ def __init__(
         answer_store: AnswerStore,
         list_store: ListStore,
         progress_store: ProgressStore,
-        metadata: ImmutableDict,
+        metadata: Mapping,
```

Comment:
Worth doing the first level. Need to import `Any`.
```suggestion
        metadata: Mapping[str, Any],
```